### PR TITLE
Drop python3.7 and add support for python3.11

### PR DESCRIPTION
**Pull request by guillemcortes**

### CODE REVIEW

File: .github/workflows/lint-python.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
       matrix:
         include:
           - os: ubuntu-latest
-            python-version: "3.8"
+            python-version: "3.11"
```
Comment (genisplaja) [inline .github/workflows/lint-python.yml:20]: Is there a particular reason for this direct jump to 3.11? Well, in the long run probably will ensure less changes as older versions of Python get deprecated...

**Review by genisplaja — APPROVED** [review]
LGTM! Python 3.7 reached end-of-life quite a long time ago, so I think it's a wise idea to remove support, while opening support for newer versions of Python. Thanksssss

File: .github/workflows/lint-python.yml (hunk 1 above)
Comment (guillemcortes) [reply .github/workflows/lint-python.yml:20]: yeah, I just thought it was good to keep up-to-date with the latest python version

Comment (genisplaja) [reply .github/workflows/lint-python.yml:20]: alright why not :) Let's move up to 3.11!

**Pull request merged**